pkg/beacon/beacon.go

```diff
@@ -74,12 +74,33 @@ func Initialize(
 	}
 
 	relayChain.OnRelayEntryRequested(func(request *event.Request) {
-		previousEntry := hex.EncodeToString(request.PreviousEntry[:])
+		currentEntryStartBlock, err := relayChain.CurrentEntryStartBlock()
+		if err != nil {
+			logger.Warningf("could not check current entry start block")
+			return
+		}
+
+		// Ignore if the block number of the received request isn't equal to
+		// the current entry start block from the chain. Most likely, there
+		// are some problems with chain synchronization and multiple events
+		// arrived at the same time but only the last should be handled.
+		if request.BlockNumber != currentEntryStartBlock.Uint64() {
+			logger.Warningf(
+				"ignoring relay entry request with block number [%v]; "+
+					"current entry start block stored on-chain is [%v]",
+				request.BlockNumber,
+				currentEntryStartBlock.Uint64(),
+			)
```

Comment:
Shouldn't we `return` here?

Reply:
Yeah, for sure. Thanks for catching this!